addons/mail/static/tests/core/record.test.js

```diff
@@ -1053,3 +1054,34 @@ test("insert with id relation keeps existing field values", async () => {
     expect(member2.channel.eq(channel1)).toBe(true);
     expect(member2.is_internal).toBe(true);
 });
+
+test("insert() different PY model having same JS model", async () => {
+    (class Thread extends Record {
+        static id = "id";
+    }).register(localRegistry);
+    patchWithCleanup(pyToJsModels, {
+        "discuss.channel": "Thread",
+        "mail.thread": "Thread",
+    });
+    const store = (await start2()).services["mail.store"];
+    const data = {
+        "discuss.channel": [
+            { id: 1, name: "General" },
+            { id: 2, name: "Sales" },
+        ],
+        "mail.thread": [
+            {
+                id: 1,
+                model: "discuss.channel",
+                module_icon: "/mail/static/description/icon.png",
+            },
+        ],
+    };
+
+    const { Thread: threads } = store.insert(data);
+    expect(threads).toHaveLength(2);
+    const general = store.Thread.get({ id: 1, model: "discuss.channel" });
+    const sales = store.Thread.get({ id: 2, model: "discuss.channel" });
+    expectRecord(sales).toBeIn(threads);
+    expectRecord(general).toBeIn(threads);
```

Comment:
`name`, `model`, `module_icon` do not contribute to test, only `id`. I would focus on just `id` so the test is minimal and straight-forward

Reply:
The actual code of Discuss uses them, but here you can just share 2 same py models using same id. This is not that useful in actual code but this is a test so KIS

Reply:
Agreed for module_icon, id+model are required (composite id)

Reply:
That's because the test is using `store_service` which contains business-level code of Discuss...
The tests in `record.test.js` should affect core record tests, without business concerns. All models should be declared, like the `Thread` you did.

Ok for now as a fix, but I would like at some point to adapt and have feature be model store rather than store_service, so that the test is unit-level and all data are explicit in test ;)